### ISSUE TITLE
Run system tests in environments that can't provision an external IP via LoadBalancer service e.g. Kind

### ISSUE THREAD

**Issue by j4mcs** [rabbitmq/cluster-operator#122]
**Is your feature request related to a problem? Please describe.**
We should run our system tests on [Kind](https://kind.sigs.k8s.io/docs/user/quick-start/). Currently we run the system tests on GKE. This incurs cost in maintaining the nodes needed for the Kubernetes cluster and the time taken to deploy resources for the tests. We are also not trying to validate that the cluster works on a particular Iaas with these tests so there is no requirement to deploy on VMs (instead of in docker). 

**Describe the solution you'd like**

- ~~Determine how to get a routable IP for clusters created in Kind~~ (Not possible in Mac)
  - ~~We use [metallb](https://github.com/pivotal/rabbitmq-for-kubernetes/blob/master/Makefile#L150) to allocate a static IP when a Service requests a Loadbalancer. This IP is not routable from the host (at least not on a Mac)~~. 
- ~~Calculate the resource requirement of the system tests given they run in parallel~~ (Our system tests use unlimeted resources)
- ~~Provision a gcloud VM that meets the tests CPU and memory requirements~~ 

Use Service type `NodePort` instead of `LoadBalancer` and access our ingress service via Kubernetes node IP. `NodePort` will allow to run system tests in Linux Kind and in environments that can't provision an external IP like vSphere Kubernetes or PKS on vSphere.

**Describe alternatives you've considered**
- For the time being it makes sense to continue using Concourse for this test. Once we've successfully resolved the technical challenges we might consider running our entire test suite in Github actions but that can be a secondary goal

- An alternative option for dealing with the LoadBalancer IP issue might be to run the system tests from within the Kind cluster and use the ingress Service internal DNS name. However, this would require rewriting the some of the test code which might be out of scope/future work

Edit: the solution has changed significantly from the original solution proposed. Changing to reflect the actual solution achieved.
